### PR TITLE
fix biachuan-7b tp

### PR DESCRIPTION
**Pull request by Sanster**
The main modifications are in the "load_weights" function.

Before:
image

After:
image

### CODE REVIEW

**Comment by LiVincent-Zhang** [comment]
Is the same reason for baichuan-13b? #530

**Comment by Sanster** [comment]
> Is the same reason for baichuan-13b? #530

Yes. I have tested it on both baichuan13b and 7b, and it can output normal output under tp.

**Comment by LiVincent-Zhang** [comment]
> > Is the same reason for baichuan-13b? #530
> 
> Yes. I have tested it on both baichuan13b and 7b, and it can output normal output under tp.

Can I use this PR directly on 13B？

File: vllm/model_executor/models/baichuan.py [hunk 1]
```diff
@@ -244,41 +251,70 @@ def forward(
...
+                num_heads = total_num_heads // tp_world_size
+                head_start = tp_rank * num_heads
+                head_end = (tp_rank + 1) * num_heads
+
+                loaded_weight = loaded_weight.view(
+                    3, total_num_heads, head_size, hidden_size
+                )
+                loaded_weight = loaded_weight[:, head_start:head_end, :, :]
+                loaded_weight = loaded_weight.reshape(-1, hidden_size)
+
```
Comment (zhuohan123) [inline vllm/model_executor/models/baichuan.py:305]: Is this part the only part that actually changes the code logic? Can you remove other format-only modifications and use `format.sh` script provided by us to re-format the code? Thanks!

**Review by zhuohan123 — CHANGES_REQUESTED** [review]
Thank you for your contribution! Can you use our official formatting script and remove other additional format changes?

File: vllm/model_executor/models/baichuan.py (hunk 1 above)
Comment (Sanster) [reply vllm/model_executor/models/baichuan.py]: Hi, I have already modified the content of the PR and removed the invalid format part.

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thank you for your contribution!

**Pull request merged**